### PR TITLE
adjust provisioning java variables to slow down provisioning FLOC-4026

### PR DESCRIPTION
**Pull request by Azulinho**
This is an attempt to slow down the rate at which Jenkins will spin up new slaves when the build Queue is high.
It should also prevent a situation where the first multijob to be scheduled on a jenkins master without any online slaves won't produce a number of failed slaves during the provisioning process.

https://clusterhq.atlassian.net/secure/attachment/13201/change%20with%20the%20change%20applied%20.png

### CODE REVIEW

File: roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml [hunk 1]
```diff
@@ -263,12 +263,12 @@
 - name: increase jenkins JVM heap-size
```
Comment (zubron) [inline roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml]: This task should be renamed as it's now doing more than just increasing the heap size.

File: roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml [hunk 2]
```diff
@@ -263,12 +263,12 @@
...
   lineinfile:
     dest=/etc/sysconfig/jenkins
-    line='JENKINS_JAVA_OPTIONS="-Djava.awt.headless=true -Xmx8192m -Dorg.eclipse.jetty.server.Request.maxFormContentSize=5000000 "'
+    line='JENKINS_JAVA_OPTIONS="-Dhudson.model.LoadStatistics.decay=0.9 -Dhudson.slaves.NodeProvisioner.MARGIN=5 -Dhudson.slaves.NodeProvisioner.MARGIN0=0.1 -Djava.awt.headless=true -Xmx8192m -Dorg.eclipse.jetty.server.Request.maxFormContentSize=5000000 "'
     state=present
   tags: ['jenkins', 'jenkins_config', 'jenkins_jvm']
   notify:
```
Comment (zubron) [inline roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml]: Can we remove this empty notify?

File: roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml [hunk 3]
```diff
@@ -263,12 +263,12 @@
...
   notify:
-    - safe-restart jenkins
+    - restart jenkins
```
Comment (zubron) [inline roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml]: :question: We have one other task in this file that results in a full restart of Jenkins. The name for that task includes `notifies restart`. Should we include that in the name of this task?

**Comment by zubron** [comment]
Am I right in thinking that these options will favour under-provisioning of nodes? In the graph, it still seems like there was quite a gap between busy executors and available ones but that's probably because there wasn't a steady stream of jobs being added to the queue. Do you have a graph to show what the node provisioning looked like before?

Otherwise, I think this is a good addition :+1:  We can always review the provisioning after a while. Please address the comments and merge.

File: roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml [hunk 4]
```diff
@@ -263,12 +263,12 @@
 - name: increase jenkins JVM heap-size
   lineinfile:
     dest=/etc/sysconfig/jenkins
-    line='JENKINS_JAVA_OPTIONS="-Djava.awt.headless=true -Xmx8192m -Dorg.eclipse.jetty.server.Request.maxFormContentSize=5000000 "'
+    line='JENKINS_JAVA_OPTIONS="-Dhudson.model.LoadStatistics.decay=0.9 -Dhudson.slaves.NodeProvisioner.MARGIN=5 -Dhudson.slaves.NodeProvisioner.MARGIN0=0.1 -Djava.awt.headless=true -Xmx8192m -Dorg.eclipse.jetty.server.Request.maxFormContentSize=5000000 "'
```
Comment (zubron) [inline roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml]: Can you add a comment to the description of these values or that thread that discusses them?

File: roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml (hunk 2 above)
Comment (Azulinho) [inline roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml]: done

File: roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml (hunk 3 above)
Comment (Azulinho) [inline roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml]: fine by me

File: roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml (hunk 4 above)
Comment (Azulinho) [inline roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml]: done

**Comment by Azulinho** [comment]
before the change, see here:
https://clusterhq.atlassian.net/secure/attachment/13202/before-change.png
after the change:
https://clusterhq.atlassian.net/secure/attachment/13201/change%20with%20the%20change%20applied%20.png

File: roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml (hunk 1 above)
Comment (Azulinho) [inline roles/local.Azulinho.azulinho-jenkins-server/tasks/main.yaml]: done

**Comment by Azulinho** [comment]
Thanks a lot brrrrrrridge

**Pull request merged**